src/components/data/DataTable.tsx

```diff
@@ -59,6 +65,25 @@ type ComponentProps<Row> = {
    */
   onConfirmRow?: (r: Row) => void;
 
+  /** Currently active order */
+  order?: Order<keyof Row>;
+
+  /**
+   * Callback when active order changes, by clicking a column header.
+   * When a header is clicked, if that's not the active order, it is set with
+   * order='ascending'.
+   * If the active header is clicked consecutively, direction rotates from
+   * 'ascending' to 'descending', and from 'descending' to no-active-order.
+   */
+  onOrderChange?: (order: Order<keyof Row>) => void;
+
+  /**
+   * Columns that can be used to order the table. Ignored if `onOrderChange` is
+   * not provided.
+   * Defaults to all columns.
+   */
+  orderableColumns?: Array<keyof Row>;
```

Comment:
We don't usually use the `Array<>` notation to type arrays, but I think `Array<keyof Row>` is easier to read than `(keyof Row)[]`

Reply:
I agree. I usually use `Array` too in such cases.